File: slack-reader/src/main/java/dev/number6/slackreader/dagger/AmazonSnsModule.java
package dev.number6.slackreader.dagger;

import com.amazonaws.services.sns.AmazonSNS;
import com.amazonaws.services.sns.AmazonSNSClientBuilder;
import dagger.Module;
import dagger.Provides;

@Module
public class AmazonSnsModule {

    @Provides
    public AmazonSNS amazonSNS() {
        return AmazonSNSClientBuilder.defaultClient();
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/dagger/AmazonSnsModule.kt
package dev.number6.slackreader.dagger

import com.amazonaws.services.sns.AmazonSNS
import com.amazonaws.services.sns.AmazonSNSClientBuilder
import dagger.Module
import dagger.Provides

@Module
class AmazonSnsModule {
    @Provides
    fun amazonSNS(): AmazonSNS {
        return AmazonSNSClientBuilder.defaultClient()
    }
}